compiler/src/middle_end/analyze_purity.re

```diff
@@ -121,16 +122,7 @@ let rec analyze_comp_expression =
           args,
         );
       List.for_all(x => x, arg_purities);
-    | [@implicit_arity] CRecord(_, args) =>
-      let arg_purities =
-        List.map(
-          ((_, arg)) => {
-            analyze_imm_expression(arg);
-            imm_expression_purity_internal(arg);
-          },
-          args,
-        );
-      List.for_all(x => x, arg_purities);
+    | CRecord(_) => false
```

Comment:
It would be great to return true here if the prior conditions were met _and_ all of the fields are immutable. I'll open an issue once this is merged.